src/polymer/declaration-property-handlers.ts

```diff
@@ -16,15 +16,48 @@ import * as estree from 'estree';
 
 import * as astValue from '../javascript/ast-value';
 import {JavaScriptDocument} from '../javascript/javascript-document';
-import {Severity} from '../warning/warning';
+import {ScannedBehaviorAssignment} from '../polymer/behavior';
+import {Severity, Warning} from '../warning/warning';
 
 import {analyzeProperties} from './analyze-properties';
 import {ScannedPolymerElement} from './polymer-element';
 
+export type BehaviorAssignmentOrWarning = {
+  kind: 'warning',
+  warning: Warning
+} | {kind: 'behaviorAssignment', assignment: ScannedBehaviorAssignment};
+
+export function getBehaviorAssignmentOrWarning(
+    argNode: estree.Node,
+    document: JavaScriptDocument): BehaviorAssignmentOrWarning {
+  const behaviorName = astValue.getIdentifierName(argNode);
+  if (!behaviorName) {
+    return {
+      kind: 'warning' as 'warning',
+      warning: {
+        code: 'could-not-determine-behavior-name',
+        message:
+            `Could not determine behavior name from expression of type ${argNode
+                .type}`,
+        severity: Severity.WARNING,
+        sourceRange: document.sourceRangeForNode(argNode)!
+      }
+    };
+  }
+  return {
+    kind: 'behaviorAssignment' as 'behaviorAssignment',
```

Comment:
It was needed before I declared the BehaviorAssignmentOrWarning type explicitly. Not needed anymore, removed.